daemon/container_operations.go

```diff
@@ -73,6 +73,9 @@ func (daemon *Daemon) buildSandboxOptions(cfg *config.Config, container *contain
 	} else if len(cfg.DNSOptions) > 0 {
 		sboxOptions = append(sboxOptions, libnetwork.OptionDNSOptions(cfg.DNSOptions))
 	}
+	if cfg.Features["windows-no-dns-proxy"] {
```

Comment:
```suggestion
	if runtime.GOOS == "windows" && cfg.Features["windows-no-dns-proxy"] {
```

Should it be a validation error (or a daemon warning log) to have that feature set in a non-Windows daemon config? Do we have any existing validation for configured feature flags, e.g. to catch typos or removed flags?

Reply:
> Should it be a validation error (or a daemon warning log) to have that feature set in a non-Windows daemon config?

Now, this happens ...

```
# /usr/local/bin/dockerd
unable to configure the Docker daemon with file /etc/docker/daemon.json: feature option 'windows-no-dns-proxy' is only available on Windows
```

> Do we have any existing validation for configured feature flags, e.g. to catch typos or removed flags?

I don't think so.